### PR TITLE
Feature/pdct 1499 add a limit to how many documents can be ingested in one go

### PR DESCRIPTION
**Pull request by annaCPR**
# Description
- max 1000 documents can be saved in one ingest - any more are ignored
- ingest is idempotent so the same json file can be used to ingest any remaining documents

Please include:

- a summary of the changes
- links to any related issue/ticket
- any additional relevant motivation and context
- details of any dependency updates that are required for this change

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [x] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/service/ingest.py [hunk 1]
```diff
@@ -28,6 +31,8 @@
     IngestFamilyDTO,
 )
 
+DOCUMENT_INGEST_LIMIT = 1000
```
Comment (diversemix) [inline app/service/ingest.py:34]: please can we make this a parameter - so we can pass in via the cli ... we will need to alter this ... not an issue for this PR so let's raise a ticket

Comment (annaCPR) [reply app/service/ingest.py:34]: Good call. And actually this will make testing easier. I'll raise the ticket

File: app/service/ingest.py [hunk 2]
```diff
@@ -92,16 +118,15 @@ def save_families(
...
-        # collection.validate(collections, db)
-        dto = IngestFamilyDTO(
-            **fam, corpus_import_id=corpus_import_id
-        ).to_family_create_dto(corpus_import_id)
-        geo_ids = []
-        for geo in dto.geography:
-            geo_ids.append(geography.get_id(db, geo))
-        import_id = family_repository.create(db, dto, geo_ids, org_id)
-        family_import_ids.append(import_id)
+        if not _exists_in_db(Family, fam["import_id"], db):
```
Comment (diversemix) [inline app/service/ingest.py:121]: neat :+1:

File: app/service/ingest.py [hunk 3]
```diff
@@ -38,6 +43,26 @@ class IngestEntityList(str, Enum):
...
 
 
+class BaseModel(DeclarativeMeta):
+    import_id: str
+
+
+T = TypeVar("T", bound=BaseModel)
+
+
+def _exists_in_db(entity: Type[T], import_id: str, db: Session) -> bool:
```
Comment (diversemix) [inline app/service/ingest.py:53]: neat :+1:

**Review by diversemix — APPROVED** [review]
lgtm

**Pull request merged**